ngraph/src/ngraph/op/variadic_split.cpp

```diff
@@ -142,3 +145,125 @@ shared_ptr<Node> op::v1::VariadicSplit::clone_with_new_inputs(const OutputVector
+                                  in->get_element_type().size());
+
+        return true;
+    }
+
+    bool evaluate_variadic_split(const HostTensorPtr& data_tensor,
+                                 const HostTensorPtr& axis_tensor,
+                                 const HostTensorPtr& split_lengths_tensor,
+                                 const HostTensorVector& outputs,
+                                 const Node* split_node)
+    {
+        int64_t axis;
+        switch (axis_tensor->get_element_type())
+        {
+        case element::Type_t::i32: axis = read_vector<int32_t>(axis_tensor)[0]; break;
+        case element::Type_t::i64: axis = read_vector<int64_t>(axis_tensor)[0]; break;
+        case element::Type_t::u64:
+            axis = static_cast<int64_t>(read_vector<uint64_t>(axis_tensor)[0]);
+            break;
+        default:
+            NODE_VALIDATION_CHECK(split_node,
+                                  false,
+                                  "Not supported axis type: ",
+                                  axis_tensor->get_element_type(),
+                                  " during evaluate Split:v1");
+            break;
+        }
+        axis = ngraph::normalize_axis(split_node, axis, data_tensor->get_partial_shape().rank());
+
+        std::vector<int64_t> split_lengths;
+        switch (split_lengths_tensor->get_element_type())
+        {
+        case element::Type_t::i32:
+        {
+            const auto split_lengths_i32 = read_vector<int32_t>(split_lengths_tensor);
+            split_lengths =
+                std::vector<int64_t>(std::begin(split_lengths_i32), std::end(split_lengths_i32));
+            break;
+        }
+        case element::Type_t::i64:
+        {
+            const auto split_lengths_i64 = read_vector<int64_t>(split_lengths_tensor);
+            split_lengths =
+                std::vector<int64_t>(std::begin(split_lengths_i64), std::end(split_lengths_i64));
+            break;
+        }
+        case element::Type_t::u64:
+        {
+            const auto split_lengths_u64 = read_vector<uint64_t>(split_lengths_tensor);
+            split_lengths =
+                std::vector<int64_t>(std::begin(split_lengths_u64), std::end(split_lengths_u64));
+            break;
+        }
+        default:
+            NODE_VALIDATION_CHECK(split_node,
+                                  false,
+                                  "Not supported split lengths type: ",
+                                  split_lengths_tensor->get_element_type(),
+                                  " during evaluate Split:v1");
+            break;
+        }
+
+        const auto data_shape = data_tensor->get_shape();
+        const auto neg_one = std::find(std::begin(split_lengths), std::end(split_lengths), -1);
+        if (neg_one != std::end(split_lengths)) // negative length set
+        {
+            int64_t sum_of_known_splits = 0;
+            std::for_each(std::begin(split_lengths),
```

Comment:
Since there can be only one split equal to `-1` you can just accumulate all splits and add `1` to the result. There's no need for any lambda in this case.

Reply:
good point, done